fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/NormalizePredicateRule.java

```diff
@@ -130,27 +135,74 @@ public ScalarOperator visitBetweenPredicate(BetweenPredicateOperator predicate,
     @Override
     public ScalarOperator visitCompoundPredicate(CompoundPredicateOperator predicate,
                                                  ScalarOperatorRewriteContext context) {
+        if (predicate.isAnd() || predicate.isOr()) {
+            ScalarOperator newTree = getOptimizedCompoundTree(predicate);
```

Comment:
suggest to make getOptimizedCompoundTree return Optional<ScalarOperator> type, if so, this functions can be rewritten more succinctly.
```java
return getOptimizedCompoundTree(predicate).orElse(predicate);
```